File: Docs/VB/Data/Commands/ContactCommands.vb
Imports System
Imports DevExpress.DevAV.Chat.Model


Namespace DevExpress.DevAV.Chat.Commands

    Public MustInherit Class ContactCommand
        Public ReadOnly Property Contact() As Contact
    End Class
    '
    Public Class AddMessage
        Inherits ContactCommand

        Public Sub New(ByVal contact As Contact)
            MyBase.New(contact)
        End Sub
        Public ReadOnly Property Message() As String
        Public ReadOnly Property Sent() As DateTime
    End Class
    Public Class ReadMessages
        Inherits ContactCommand

        Public Sub New(ByVal contact As Contact)
            MyBase.New(contact)
        End Sub
    End Class
End Namespace

File: Docs/CS/Data/Commands/ContactCommands.cs
using System;
using DevExpress.DevAV.Chat.Model;


namespace DevExpress.DevAV.Chat.Commands {

    public abstract class ContactCommand {
        public Contact Contact {
            get;
        }
    }
    //
    public class AddMessage : ContactCommand {
        public AddMessage(Contact contact)
            : base(contact) {
        }
        public string Message {
            get;
        }
        public DateTime Sent {
            get;
        }
    }
    public class ReadMessages : ContactCommand {
        public ReadMessages(Contact contact)
            : base(contact) {
        }
    }
}